modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/BasicIndexTest.java

```diff
@@ -1518,6 +1521,46 @@ public void testCreateLuceneIndex() throws Exception {
 
         assertNull(GridTestUtils.getFieldValue(tblDesc1, "luceneIdx"));
     }
+    
+    /**
+     * Checks that part of the composite key assembled in BinaryObjectBuilder can pass the validation correctly
+     if you specify Object type when creating the index.
+     */
+    @Test
+    public void testCacheSecondaryCompositeIndex() throws Exception {
```

Comment:
Can you, please, fix formatting and javadoc.

Reply:
Hope I fixed it, thanks.